lib/pure/strformat.nim

```diff
@@ -226,6 +226,67 @@ template callFormatOption(res, arg, option) {.dirty.} =
   else:
     format($arg, option, res)
 
+proc impl(pattern: NimNode, shouldUnescape: bool): NimNode =
```

Comment:
I had to move this above the runableExamples so the diff is huge. The only actual change was adding the makeStrLit template and calling on lines 252 and 285